### PR TITLE
Alpha Dropout

### LINKED ISSUE #10612: Implement components for "Self-normalizing networks"
Hochreiter's group has recently come up with a new dropout technique and activation function in a recent paper (https://arxiv.org/abs/1706.02515). Presented experiments demonstrate that these lead to better learning in standard feed forward networks. 

I would like to implement these components in TensorFlow. Creating this issue to gauge the community's interest level in such components. Feedback will be extremely helpful.

### PR DESCRIPTION
**Pull request by AnishShah**
Fixes #10612

### CODE REVIEW

**Comment by ebrevdo** [comment]
New functions like this should go into contrib.  I think we have a tf.contrib.nn?  I'd put it there.

**Comment by AnishShah** [comment]
The documentation for `tf.contrib.nn` says that "Module for deprecated ops in tf.nn.". Is that the correct place? or `tf.contrib.layers`?

**Comment by AnishShah** [comment]
@ebrevdo BTW, did you mean to remove this function from `nn_ops.py` and add to `contrib` or add to both places?

**Comment by ebrevdo** [comment]
I meant to move it there.  And you can remove the deprecated word of the
doc... New nn ops can go there too.

**Comment by vrv** [comment]
ping for @AnishShah !

**Comment by AnishShah** [comment]
I will update this in a day. Sorry for the delay.

**Comment by AnishShah** [comment]
I have updated the PR. Please review.

**Comment by thesuperzapper** [comment]
What's happening here?

File: tensorflow/contrib/nn/python/ops/alpha_dropout_test.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  def testAlphaDropout(self):
+    x_dim, y_dim = 40, 30
+    for keep_prob in [0.1, 0.5, 0.8]:
+      with self.test_session():
+        t = random_ops.random_normal([x_dim, y_dim])
+        output = alpha_dropout(t, keep_prob)
+        self.assertEqual([x_dim, y_dim], output.get_shape())
+        t_mean, t_std = nn_impl.moments(t, axes=[0, 1])
+        output_mean, output_std = nn_impl.moments(t, axes=[0, 1])
```
Comment (vrv) [inline tensorflow/contrib/nn/python/ops/alpha_dropout_test.py]: did you mean 'output' here instead of 't?

**Comment by AnishShah** [comment]
@vrv Sorry about that. I have changed that.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by AnishShah** [comment]
The build was aborted. Can you run the tests again?

**Comment by vrv** [comment]
The build was aborted because the sanity checks failed -- can you look at the logs here to see what went wrong?

https://ci.tensorflow.org/job/tensorflow-pull-requests-sanity/5565/consoleFull

**Comment by AnishShah** [comment]
There was some problem with BUILD file. I have fixed it. Please run the tests again.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by AnishShah** [comment]
`//tensorflow/contrib/opt:drop_stale_gradient_optimizer_test              FAILED in 4.8s` - This test failed for Linux CPU Tests 

I don't think this is related to my changes.

**Review by vrv — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please
